pkg/openstack/instances.go

```diff
@@ -625,5 +664,64 @@ func getAttachedInterfacesByID(client *gophercloud.ServiceClient, serviceID stri
 		return interfaces, err
 	}
 
+	// Check if trunk ports are attached
+	var subports []trunks.Subport
+
+	listOpts := trunks.ListOpts{}
+	allPages, err := trunks.List(network, listOpts).AllPages()
+	if err != nil {
+		klog.Errorf("Failed to list trunks: %v", err)
+		return interfaces, nil
```

Comment:
I'm wondering why the errors below are all ignored?

Additionally, I'd rather put the logic here into a separate function.

Reply:
(1) The errors were not ignored in the initial pull request. After rebasing to pick up the code re-factor in openstack.go and instances.go, the openlab/cloud-provider-openstack-acceptance-test-e2e-conformance test starts to fail. By ignoring the errors, the CI test passes. I am unable to make a sense of it.
(2) Creating a separate function sounds good, I will do.

Reply:
> By ignoring the errors, the CI test passes

Ignoring the errors doesn't sound a good approach, if the CI failed just because we catch the error here, there must be bugs. So please add the errors back and let's see what the failure will be.

Reply:
Reverted back to return error in case of trunk API failures.